2.rb

```diff
@@ -0,0 +1,7 @@
+arr = "abcavdfa"
+freq=Hash.new  
+freq.default=0
```

Comment:
What does this line do ??
Can you please explain

Reply:
Sure. arr is a string, and I have initialized a hash map in the following line. and in the third line, I have set the key values as default 0 because I need to calculate the frequency. So setting the key value to 0 before calculating.